DigitalOffice/Services/UserService/Broker/Consumers/GetUserInfoConsumer.cs

```diff
@@ -0,0 +1,56 @@
+﻿using System.Threading.Tasks;
+using Broker.Requests;
+using Broker.Responses;
+using LT.DigitalOffice.Kernel.Broker;
+using LT.DigitalOffice.UserService.Database.Entities;
+using LT.DigitalOffice.UserService.Mappers.Interfaces;
+using LT.DigitalOffice.UserService.Repositories.Interfaces;
+using MassTransit;
+using Microsoft.AspNetCore.Mvc;
+
+namespace LT.DigitalOffice.UserService.Broker.Consumers
+{
+    /// <summary>
+    /// Consumer for getting information about the user
+    /// </summary>
+    public class GetUserInfoConsumer : IConsumer<IGetUserInfoRequest>
+    {
+        private readonly IRequestClient<IGetUserPositionRequest> client;
+        private readonly IUserRepository repository;
+        private readonly ITwoModelsMapper<DbUser,IUserPositionResponse, object> mapper;
+
+        public GetUserInfoConsumer(
+            [FromServices] IUserRepository repository,
+            [FromServices] IRequestClient<IGetUserPositionRequest> client,
+            [FromServices] ITwoModelsMapper<DbUser,IUserPositionResponse, object> mapper)
+        {
+            this.repository = repository;
+            this.client = client;
+            this.mapper = mapper;
+        }
+
+        public async Task Consume(ConsumeContext<IGetUserInfoRequest> context)
+        {
+            var response= await client.GetResponse<IOperationResult<IUserPositionResponse>>(
+                new
+                {
+                    context.Message.UserId
+                });
+            if (!response.Message.IsSuccess)
+            {
+                await context.RespondAsync(response.Message);
+            }
+            else
+            {
+                var dbUser = repository.GetUserInfoById(context.Message.UserId);
```

Comment:
What if the repository throws an exception? Do we want to report the exception to the sender?

Reply:
Fixed, I forgot to catch an exception here